internal/controller/datamovement_controller.go

```diff
@@ -460,6 +498,78 @@ func (r *DataMovementReconciler) Reconcile(ctx context.Context, req ctrl.Request
 	return ctrl.Result{}, nil
 }
 
+func parseDcpProgress(line string, cmdStatus *nnfv1alpha1.NnfDataMovementCommandStatus) error {
+	match := progressRe.FindStringSubmatch(line)
+	if len(match) > 0 {
+		progress, err := strconv.Atoi(match[1])
+		if err != nil {
+			return fmt.Errorf("failed to parse progress output: %w", err)
+		}
+		progressInt32 := int32(progress)
+		cmdStatus.ProgressPercentage = &progressInt32
+	}
+
+	return nil
+}
+
+// Go through the list of dcp stat regexes, parse them, and put them in their appropriate place in cmdStatus
+func parseDcpStats(line string, cmdStatus *nnfv1alpha1.NnfDataMovementCommandStatus) error {
```

Comment:
I'm all ears if there's a better way to do this

Reply:
You just need to add an option to dcp to output in json.